### PR TITLE
Repro #25988: Cannot reference nested question as native subquery

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Reproduces #25988 

### How to test this manually?
- `yarn test-cypress-open`
- `frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js`
- Replace `it.skip()` with `it.only()` to run the test in isolation
- The test should fail until the related issue is fixed

### Additional notes:
- Once the issue is fixed, please remove the `.skip` part (unskip the test completely)
- Make sure the test is passing and
- Merge it together with the fix

### Screenshots:
image

### CODE REVIEW

File: frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js [hunk 1]
```diff
@@ -263,4 +267,33 @@ describe("scenarios > question > native subquery", () => {
...
+        "source-table": "card__1",
+        limit: 2,
+      },
+    };
+
+    cy.createQuestion(questionDetails).then(
+      ({ body: { id: nestedQuestionId } }) => {
+        const tagID = `#${nestedQuestionId}`;
+
+        startNewNativeQuestion();
```
Comment (ranquild) [inline frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js:284]: There is existing `openNativeEditor` helper that does the same

File: frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js [hunk 2]
```diff
@@ -263,4 +267,33 @@ describe("scenarios > question > native subquery", () => {
...
+        SQLFilter.enterParameterizedQuery(`SELECT * FROM {{${tagID}`);
+
+        cy.findByTestId("sidebar-header-title")
+          .invoke("text")
+          .should("eq", questionDetails.name);
+
+        runNativeQuery({
```
Comment (ranquild) [inline frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js:291]: Could we rely on some results present, and not on the backend not error-ing?

File: frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js:284]: It doesn't.
The other one uses the UI and is much slower. This one generates a base64 string from an ad-hoc query and goes straight to the native editor.

File: frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js (hunk 2 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js:291]: Absolutely.
That's why line 295 is there for.

Comment (ranquild) [reply frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js:291]: I still don't think we should assert for a dataset response here

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js:291]: I'm happy to remove that assertion but am genuinely interested to learn why do you think that?
Are there any downsides to this approach that I'm not seeing?

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/native/native_subquery.cy.spec.js]: Removed, btw
https://github.com/metabase/metabase/pull/28135/commits/984775a4b5bceb1cbfa1d22509a6466ebac93535

File: frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js [hunk 3]
```diff
@@ -43,12 +43,14 @@ export function openNativeEditor({
  * Executes native query and waits for the results to load.
  * Makes sure that the question is not "dirty" after the query successfully ran.
  */
-export function runNativeQuery({ wait = true } = {}) {
+export function runNativeQuery({ wait = true, callback } = {}) {
```
Comment (ranquild) [inline frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js:46]: @nemanjaglumac this callback is not used, could you revert changes here?

**Review by ranquild — APPROVED** [review]
Please take a look at the last commend

**Pull request merged**